### ISSUE TITLE
Feature request: snapshot retention

### ISSUE THREAD

**Issue by TroLoos** [BassT23/Proxmox#110]
Hi, first of all - great and very complete script you have created! It's really useful and makes my (and ours) lives so much easier.

Would be difficult to implement config-defined backup snapshot retention?

**Comment by TroLoos** [comment]
Hi, I updated your script, works like a charm ;-). Thank you!

For future if you find time, it would be nice to have an option to define how many last snapshots to keep (for example: last 7) and delete only older ones. But it's not that critical as last snapshot before update is a very nice-to-have solution to quickly rollback.

Thanks a lot!

**Comment by TroLoos** [comment]
Ok, I did some testing and something doesn't work fully. One one Proxmox host, it goes all fine, snapshots are created, old snapshots deleted. I have one VM excluded there and it works fine.

On the other host, after updating the script, snapshots work only for the first LXC being updated:

```
[Info] Updating LXC 201 : plex

[Info] Start snaphot and/or backup
[Info] Snapshot created
[Info] delete old snapshots
```

but later, for next LXC it is like this:

```
[Info] Updating LXC 210 : frigate-lxc

[Info] Start snaphot and/or backup
[Info] Snapshot and Backup skipped by user
```

and later for the VM:

```
[Info] Updating VM 103 : StarrApps

[Info] Start snaphot and/or backup
[Info] Snapshot and Backup skipped by user
```

Even if I wanted to behave like this, I couldn't - snapshots are general setting for all VM/LXC so there's no chance this could be a configuration issue, right?

**Comment by BassT23** [comment]
> Hi, I updated your script, works like a charm ;-). Thank you!
> 
> For future if you find time, it would be nice to have an option to define how many last snapshots to keep (for example: last 7) and delete only older ones. But it's not that critical as last snapshot before update is a very nice-to-have solution to quickly rollback.
> 
> Thanks a lot!

implemented in latest beta

**Comment by TroLoos** [comment]
It looks good, with snapshot retention, and snapshot I confirm that are created for 1st and later lxc/vm.

Now, the script stops when doing ssh update of vm with message:

--- Searching for extra updates ---
chmod: cannot access '/update-extras.sh': No such file or directory
Error during Update --- Exit Code: 1

Can you check what might be a reason?

**Comment by TroLoos** [comment]
> the other "problem" I also see in first run, but on the next time, its gone. So hopefully its only an "initial problem" ;)

Hi @BassT23, I've been running your script weekly and it works just great, it's been a really great work on your side and a lot of help for us/users.

The only issue I've been seeing right now is that snapshots are created only for first lxc/vm on the host, the rest is still 'Snapshot and Backup skipped by user'. If you find time to look into this, it would be great. Otherwise, I will still use this script anyway ;-).

**Comment by TroLoos** [comment]
I looked at the code and it looks like the problem is here:
`for SNAPSHOT in $LIST; do
pct delsnapshot "$CONTAINER" Update"$SNAPSHOT"`
You use the same name for iteration as the SNAPSHOT variable in config file, and it gets overwritten here.

**Comment by BassT23** [comment]
@TroLoos, you could be right.
Hopefully fixed with v4.0.6 :)
THX for the hint ;)

**Comment by TroLoos** [comment]
> EDIT: test is running now and look good ;)

Yes, I confirm - it all works very nicely now. Thank you, this problem is now resolved.
